graylog2-web-interface/src/views/components/widgets/CopyToDashboardForm.tsx

```diff
@@ -16,23 +16,25 @@
  */
 import React, { useState, useCallback, useEffect, useRef } from 'react';
 
-import { Modal, ListGroup, ListGroupItem } from 'components/bootstrap';
+import { Modal, ListGroup, ListGroupItem, Input } from 'components/bootstrap';
 import { PaginatedList, SearchForm, ModalSubmit, Spinner } from 'components/common';
 import useDashboards from 'views/components/dashboard/hooks/useDashboards';
 import type { SearchParams } from 'stores/PaginationTypes';
 
 type Props = {
   activeDashboardId?: string,
   onCancel: () => void,
-  onSubmit: (selectedDashboardId: string | undefined | null) => Promise<void>,
+  onCopyToDashboard: (selectedDashboardId: string | undefined | null) => Promise<void>,
   submitButtonText: string,
   submitLoadingText: string,
+  onCreateNewDashboard?: () => Promise<void>,
```

Comment:
Currently this function does not seem to return a promise.

Reply:
We need it. in the end, we have part of the code 

```
    submitHandler().then(() => {
      if (isMounted.current) {
        setIsSubmitting(false);
      }
    });
```

where `submitHandler` 

`submitHandler = async () => (createNewDashboard ? onCreateNewDashboard() : onCopyToDashboard(selectedDashboard));`

handle copying widget to dashboard or creating a new one